front/src/applications/operationalStudies/hooks/useSimulationResults.ts

```diff
@@ -19,12 +32,55 @@ const useSimulationResults = (infraId: number): SimulationResults | undefined =>
   const electricalProfileSetId = useSelector(getOperationalStudiesElectricalProfileSetId);
   const selectedTrainId = useSelector(getSelectedTrainId);
 
-  const train = useSelectedTrain();
+  const timetableItem = useSelectedTimetableItem();
+
+  const train = useMemo(() => {
+    if (!selectedTrainId || !timetableItem) return undefined;
+    if (!isPacedTrainResponseWithPacedTrainId(timetableItem)) {
+      return timetableItem;
+    }
+
+    if (isTrainScheduleId(selectedTrainId)) {
+      throw new Error(`trainId ${selectedTrainId} should be a occurrence id`);
+    }
+
+    const exception = findExceptionWithOccurrenceId(timetableItem.exceptions, selectedTrainId);
+
+    // handle exceptions
+    if (exception) {
+      return {
+        ...extractOccurrenceDetailsFromPacedTrain(timetableItem, exception),
+        id: selectedTrainId,
+      };
+    }
+
+    // handle regular indexed occurrence
+    const selectedOccurrenceIndex = extractOccurrenceIndexFromOccurrenceId(selectedTrainId);
+    const pacedTrainIntervalInMs = Duration.parse(timetableItem.paced.interval).ms;
+    const selectedOccurrenceStartTime: string = dayjs(timetableItem.start_time)
+      .add(selectedOccurrenceIndex * pacedTrainIntervalInMs, 'ms')
+      .toISOString();
+
+    return {
+      ...timetableItem,
+      id: selectedTrainId,
+      start_time: selectedOccurrenceStartTime,
+    };
+  }, [timetableItem]);
+
+  const exception = useMemo(() => {
+    if (!selectedTrainId || !timetableItem || !isPacedTrainResponseWithPacedTrainId(timetableItem))
+      return undefined;
+    if (isTrainScheduleId(selectedTrainId))
+      throw new Error(`trainId ${selectedTrainId} should be a occurrence id`);
```

Comment:
as you use this snippet twice, can this be extract in a small function ? to make it testable ?

Reply:
Hum I tried, but it's not really possible because the checks/throw errors are needed for typescript.
If I extract them in a small function, then typescript does not know anymore than `selectedTrainId` is an occurrence id.

We should keep it like this if it's fine for you